traits/has_traits.py

```diff
@@ -2047,7 +2035,40 @@ def class_trait_view_elements(cls):
         The returned object can be used to access all the view elements
         associated with the class.
         """
-        return cls.__dict__[ViewTraits]
+        view_elements = cls.__dict__[ViewTraits]
+        if isinstance(view_elements, dict):
+            view_elements = cls._init_trait_view_elements()
+        return view_elements
+
+    @classmethod
+    def _init_trait_view_elements(cls):
+        """ Lazily Initialize the ViewElements object from a dictionary. """
+        from traitsui.view_elements import ViewElements
+
+        hastraits_bases = [
+            base for base in cls.__bases__
+            if base.__dict__.get(ClassTraits) is not None
+        ]
+        view_elements = ViewElements()
+        elements_dict = cls.__dict__[ViewTraits]
+
+        for name, element in elements_dict:
+            view_elements.content[name] = element
+
+            # Replace all substitutable view sub elements with 'Include'
+            # objects, and add the substituted items to the
+            # 'ViewElements':
+            element.replace_include(view_elements)
+
+        for base in hastraits_bases:
```

Comment:
I guess this is where we might need/want the duplication check.

Could we have a test that exercises the duplication check in the original code?

Reply:
I don't think you can have duplicates.